pkl-core/src/main/java/org/pkl/core/util/IoUtils.java

```diff
@@ -365,13 +365,13 @@ public static URI relativize(URI uri, URI base) {
       return uri;
     }
 
-    var basePath = Path.of(base.getRawPath());
+    var basePath = Path.of(base.getPath());
     if (!base.getRawPath().endsWith("/")) basePath = basePath.getParent();
-    var resultPath = basePath.relativize(Path.of(uri.getRawPath()));
+    var resultPath = basePath.relativize(Path.of(uri.getPath()));
 
     try {
       return new URI(
-          null, null, null, -1, resultPath.toString(), uri.getRawQuery(), uri.getRawFragment());
+          null, null, null, -1, resultPath.toString(), uri.getQuery(), uri.getFragment());
```

Comment:
This was incorrect; using the raw API means we end up double encoding things. So, for example, a path value `/foo bar` gets turned into URI `/foo%2520bar`, when it should actually be `/foo%20bar`

Reply:
Good catch